File: java/src/hashing/LC_202_HappyNumber.java
package hashing;
// https://leetcode.com/problems/happy-number/

import java.util.HashSet;

public class LC_202_HappyNumber {

    public boolean isHappy(int n) {
        HashSet<Integer> seen = new HashSet<>();
        while (n != 1 && !seen.contains(n)) {
            int digitsSum = 0;
            while (n > 0) {
                digitsSum += Math.pow(n % 10, 2);
                n = n / 10;
            }
            n = digitsSum;
        }
        return n == 1 ? true : false;
    }
}

File: python/hashing/LC_202_HappyNumber.py
# https://leetcode.com/problems/happy-number/
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1 and n not in seen:
            seen.add(n)
            digits_sum = 0
            while n > 0:
                digits_sum += (n % 10) ** 2
                n = n / 10
            n = digits_sum
        return True if n == 1 else False
